src/ngraph/serializer.hpp

```
@@ -62,3 +62,43 @@ namespace ngraph
     /// Option may be enabled by setting the environment variable NGRAPH_SERIALIZER_OUTPUT_SHAPES
     void set_serialize_output_shapes(bool enable);
 }
+
+#ifndef NGRAPH_JSON_ENABLE
+// Rather than making every reference to the serializer conditionally compile here we just
+// provide some null stubs to resolve link issues
+// The `inline` is so we don't get multiple definitions of function
+std::string inline ngraph::serialize(std::shared_ptr<ngraph::Function> func, size_t indent)
+{
+    return "";
```

Should we throw an error instead of silent failure?